core/controllers/profile.py

```diff
@@ -374,27 +374,32 @@ class UserInfoHandler(base.BaseHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
-    @acl_decorators.require_user_id
+    @acl_decorators.open_access
     def get(self):
         """Handles GET requests."""
-
-        user_actions = user_services.UserActionsInfo(self.user_id).actions
-        user_settings = user_services.get_user_settings(
-            self.user_id, strict=False)
-        self.render_json({
-            'is_moderator': (
-                user_services.is_at_least_moderator(self.user_id)),
-            'is_admin': user_services.is_admin(self.user_id),
-            'is_super_admin': (
-                current_user_services.is_current_user_super_admin()),
-            'is_topic_manager': user_services.is_topic_manager(self.user_id),
-            'can_create_collections': bool(
-                role_services.ACTION_CREATE_COLLECTION in user_actions),
-            'preferred_site_language_code': (
-                user_settings.preferred_site_language_code),
-            'username': user_settings.username,
-            'user_is_logged_in': True
-        })
+        if self.user_id:
+            user_actions = user_services.UserActionsInfo(self.user_id).actions
+            user_settings = user_services.get_user_settings(
+                self.user_id, strict=False)
+            self.render_json({
+                'is_moderator': (
+                    user_services.is_at_least_moderator(self.user_id)),
+                'is_admin': user_services.is_admin(self.user_id),
+                'is_super_admin': (
+                    current_user_services.is_current_user_super_admin()),
+                'is_topic_manager': (
+                    user_services.is_topic_manager(self.user_id)),
+                'can_create_collections': bool(
+                    role_services.ACTION_CREATE_COLLECTION in user_actions),
+                'preferred_site_language_code': (
+                    user_settings.preferred_site_language_code),
+                'username': user_settings.username,
+                'user_is_logged_in': True
+            })
+        else:
+            self.render_json({
+                'user_is_logged_in': False
```

Comment:
Should we be returning the other keys as well (with is_XXX = False, username = None, preferred_site_language_code = None) for parity with the logged-in case?

Reply:
Yes. I just realized by doing this we can remove other global variables also like username, isAdmin, isModerator, isTopicManager. Done!